### PR TITLE
[go_router] Added `initialExtra` param to GoRouter

### PR DESCRIPTION
**Pull request by sarbagyastha**
Adds `initialExtra` param that can use to send extra object alongside `initialLocation`.

Fixes https://github.com/flutter/flutter/issues/122780

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by chunhtai** [comment]
It looks like you need to run format and also bump the package version

**Comment by sarbagyastha** [comment]
> It looks like you need to run format and also bump the package version

@chunhtai Done

File: packages/go_router/CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
-## NEXT
+## 6.2.1
```
Comment (chunhtai) [inline packages/go_router/CHANGELOG.md:1]: this should probably be 6.3.0

File: packages/go_router/pubspec.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 name: go_router
 description: A declarative router for Flutter based on Navigation 2 supporting
   deep linking, data-driven routes and more
-version: 6.2.0
+version: 6.2.1
```
Comment (chunhtai) [inline packages/go_router/pubspec.yaml:4]: same here

**Review by chunhtai — APPROVED** [review]
LGTM

**Review by hangyujin — APPROVED** [review]
LGTM

**Pull request merged**